source/extensions/filters/http/admission_control/admission_control.cc

```diff
@@ -0,0 +1,138 @@
+#include "extensions/filters/http/admission_control/admission_control.h"
+
+#include <chrono>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "envoy/extensions/filters/http/admission_control/v3alpha/admission_control.pb.h"
+#include "envoy/grpc/status.h"
+#include "envoy/http/codes.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/server/filter_config.h"
+
+#include "common/common/assert.h"
+#include "common/common/cleanup.h"
+#include "common/common/enum_to_int.h"
+#include "common/grpc/common.h"
+#include "common/http/codes.h"
+#include "common/http/utility.h"
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/http/admission_control/evaluators/default_evaluator.h"
+#include "extensions/filters/http/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AdmissionControl {
+
+using GrpcStatus = Grpc::Status::GrpcStatus;
+
+static constexpr double defaultAggression = 2.0;
+
+AdmissionControlFilterConfig::AdmissionControlFilterConfig(
+    const AdmissionControlProto& proto_config, Runtime::Loader& runtime, TimeSource& time_source,
+    Runtime::RandomGenerator& random, Stats::Scope& scope, ThreadLocal::SlotPtr&& tls,
+    std::unique_ptr<ResponseEvaluator> response_evaluator)
+    : runtime_(runtime), time_source_(time_source), random_(random), scope_(scope),
+      tls_(std::move(tls)), admission_control_feature_(proto_config.enabled(), runtime_),
+      aggression_(
+          proto_config.has_aggression_coefficient()
+              ? std::make_unique<Runtime::Double>(proto_config.aggression_coefficient(), runtime_)
+              : nullptr),
+      response_evaluator_(std::move(response_evaluator)) {}
+
+double AdmissionControlFilterConfig::aggression() const {
+  return std::max<double>(1.0, aggression_ ? aggression_->value() : defaultAggression);
+}
+
+AdmissionControlFilter::AdmissionControlFilter(AdmissionControlFilterConfigSharedPtr config,
+                                               const std::string& stats_prefix)
+    : config_(std::move(config)), stats_(generateStats(config_->scope(), stats_prefix)) {}
+
+Http::FilterHeadersStatus AdmissionControlFilter::decodeHeaders(Http::RequestHeaderMap&, bool) {
+  if (!config_->filterEnabled() || decoder_callbacks_->streamInfo().healthCheck()) {
```

Comment:
It makes sense to ignore health checks, but can you make sure this is documented in the final docs?